crypto/src/main/java/org/cryptimeleon/incentive/crypto/proof/spend/tree/SpendDeductLeafNode.java

```diff
@@ -0,0 +1,34 @@
+package org.cryptimeleon.incentive.crypto.proof.spend.tree;
+
+import org.cryptimeleon.math.structures.rings.cartesian.RingElementVector;
+
+/**
+ * Leaf nodes that represent the actual statements of the boolean formula.
+ */
+public abstract class SpendDeductLeafNode implements SpendDeductTree {
+
+    // Name must be unique within the whole SpendDeductTree
+    private final String leafName;
+
+    /**
+     * Create a leaf node with a name that is unique within the tree.
+     *
+     * @param leafName name of the leaf node
+     */
+    public SpendDeductLeafNode(String leafName) {
+        this.leafName = leafName;
+    }
+
+    public String getLeafName() {
+        return leafName;
+    }
+
+    /**
+     * Function that determines whether this leaf node's ZKP is satisfied by the given old and new points vectors.
+     *
+     * @param pointsVector    vector containing the old token's points
+     * @param newPointsVector vector containing the points the users want for their new token
+     * @return whether the point vectors satisfy the ZKP relation
+     */
+    public abstract boolean isValidForPoints(RingElementVector pointsVector, RingElementVector newPointsVector);
```

Comment:
@JanBobolz what do you think about this api. I am not sure how we can cleverly include the newPointsVector. We somehow need to know which `newPointsVectors` satisfy a proofs relation, maybe add a second method `computeQualifyingNewPointsVector(RingElementVector pointsVector) -> newPointsVector` that given a pointsVector return a newPointsVector / a set of those? The expressiveness of our ProofsOfPartialKnowledge make modelling this quite hard.

Reply:
As discussed in today's meeting, the user won't make that decision based on what he has to prove, but rather the promotion should present several options to the user and then depending on the user's (secret) choice, the promotion computes the concrete point update on the old token.